lightning/src/ln/channelmanager.rs

```diff
@@ -6337,6 +6337,14 @@ where
 	///
 	/// Will regularly be called by the background processor.
 	pub fn process_pending_htlc_forwards(&self) {
+		static REENTRANCY_GUARD: AtomicBool = AtomicBool::new(false);
```

Comment:
Did this happen, another round of processing still underway? Also wondering if processing can be skipped accidentally.

Reply:
>That's not an accidental skip though, they will be processed as part of the next batch.

Not everyone may use our background processor. I think it is the expectation that `process_pending_htlc_forwards` does what it says? Or otherwise return an error perhaps.

>What do you mean by that?

The persistence lock is released and then the atomic bool is reset. In between, another forward may come in, that is then not processed because the atomic bool is still set. The same argument basically, that it isn't fully safe, unless you keep retrying.

Reply:
Ah I see indeed. But is it really necessary to have the reentry guard? If users call this a bit more often, the only thing that would happen is that they may have to wait for the lock. And waiting for the lock may happen much more often anyway, because we are now polling.